Source/WebCore/page/EventCounts.cpp

```diff
@@ -25,14 +25,133 @@
 
 #include "config.h"
 #include "EventCounts.h"
-
+#include "IDLTypes.h"
 #include "JSDOMMapLike.h"
+#include <algorithm>
+#include <initializer_list>
+#include <wtf/NeverDestroyed.h>
 
 namespace WebCore {
 
-void EventCounts::initializeMapLike(DOMMapAdapter&)
+static constexpr std::array COUNTED_EVENT_SET {
+    EventType::auxclick,
+    EventType::click,
+    EventType::contextmenu,
+    EventType::dblclick,
+    EventType::mousedown,
+    EventType::mouseenter,
+    EventType::mouseleave,
+    EventType::mouseout,
+    EventType::mouseover,
+    EventType::mouseup,
+    EventType::pointerover,
+    EventType::pointerenter,
+    EventType::pointerdown,
+    EventType::pointerup,
+    EventType::pointercancel,
+    EventType::pointerout,
+    EventType::pointerleave,
+    EventType::gotpointercapture,
+    EventType::lostpointercapture,
+    EventType::touchstart,
+    EventType::touchend,
+    EventType::touchcancel,
+    EventType::keydown,
+    EventType::keypress,
+    EventType::keyup,
+    EventType::beforeinput,
+    EventType::input,
+    EventType::compositionstart,
+    EventType::compositionupdate,
+    EventType::compositionend,
+    EventType::dragstart,
+    EventType::dragend,
+    EventType::dragenter,
+    EventType::dragleave,
+    EventType::dragover,
+    EventType::drop
+};
+
+EventCounts::EventCounts()
+{ }
+
+void EventCounts::add(EventType t)
+{
+    auto index = eventTypeIndex(t);
+    ASSERT(index);
+    ++m_counts[index.value()];
+}
+
+std::optional<size_t> EventCounts::eventTypeIndex(EventType t)
+{
+    auto it = std::ranges::find(COUNTED_EVENT_SET, t);
```

Comment:
This will do a linear search. Please sort the array and use binary search. (e.g. [ranges::lower_bound](https://en.cppreference.com/w/cpp/algorithm/ranges/lower_bound.html))